src/auth/BaseAuthApi.ts

```diff
@@ -1,14 +1,48 @@
-import { BaseAPI, Configuration as BaseConfiguration } from '../runtime';
+import { BaseAPI, Configuration as BaseConfiguration, ResponseError } from '../runtime';
 import { AddClientAuthenticationPayload, addClientAuthentication } from './clientAuthentication';
+import { Response } from 'node-fetch';
 
-export interface Configuration extends Omit<BaseConfiguration, 'baseUrl'> {
+export interface Configuration extends Omit<BaseConfiguration, 'baseUrl' | 'parseError'> {
   domain: string;
   clientId: string;
   clientSecret?: string;
   clientAssertionSigningKey?: string;
   clientAssertionSigningAlg?: string;
 }
 
+interface AuthApiErrorResponse {
+  error_description: string;
+  error: string;
+}
+
+export class AuthApiError extends Error {
+  override name = 'AuthApiError' as const;
+  constructor(public error: string, public error_description: string) {
+    super(error_description);
+  }
+}
+
+async function parseError(response: Response) {
+  // Errors typically have a specific format:
+  // {
+  //    error: 'invalid_body',
+  //    error_description: 'Bad Request',
+  // }
+
+  let data: AuthApiErrorResponse;
+
+  try {
+    data = (await response.json()) as AuthApiErrorResponse;
```

Comment:
You can't read the body of a response more than once, so would be good to give the response body in the generic error along with the headers (also, matches how we do responses).

Something like how [okta does it](https://github.com/okta/okta-sdk-nodejs/blob/26ba0a02e6e0b96467014f5125a370668416e881/src/http.js#L34):

Also, let's not add a case for the body being json but not having the error property for now (unless you know of a case?)

Also, let's make `error_description` optional, since this is the case with the oauth spec.,

You could probably do AuthApiError extends ResponseError and have the latter hold the raw body, headers etc.

Something like

```js
body = await response.text()
try {
  json = JSON.parse(body)
  return new AuthApiError(json, status, body, headers); 
} catch (_) {
   return new ResponseError(status, body, headers);
}
```